File: app/src/main/java/com/icerockdev/babenko/ui/base/fragments/ServerErrorDialogFragment.java
package com.icerockdev.babenko.ui.base.fragments;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.icerockdev.babenko.R;

/**
 * Created by Roman Babenko on 30/04/17.
 */

public class ServerErrorDialogFragment extends DialogFragment {
    public static final String DIALOG_MESSAGE_KEY = ServerErrorDialogFragment.class.getName()
            + ".DIALOG_MESSAGE_KEY";

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle arguments = getArguments();
        String message = arguments.getString(DIALOG_MESSAGE_KEY);
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        builder.setTitle(getString(R.string.request_data_fields_error))
                .setMessage(message);
        builder.setPositiveButton(android.R.string.ok, null);
        return builder.create();
    }
}

File: app/src/main/java/com/icerockdev/babenko/ui/base/fragments/ServerErrorDialogFragment.kt
package com.icerockdev.babenko.ui.base.fragments

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v7.app.AlertDialog

import com.icerockdev.babenko.R

/**
 * Created by Roman Babenko on 30/04/17.
 */
class ServerErrorDialogFragment : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val arguments = arguments
        var message: String? = context!!.getString(R.string.error_other)
        if (arguments != null) {
            message = arguments.getString(DIALOG_MESSAGE_KEY)
        }
        val builder = AlertDialog.Builder(context!!)
        builder.setTitle(getString(R.string.error_title))
                .setMessage(message)
        builder.setPositiveButton(android.R.string.ok, null)
        return builder.create()
    }

    companion object {
        private const val DIALOG_MESSAGE_KEY = "ServerErrorDialogFragment.DIALOG_MESSAGE_KEY"

        fun getDialog(error: String): ServerErrorDialogFragment {
            val arguments = Bundle()
            arguments.putString(DIALOG_MESSAGE_KEY, error)
            val serverErrorDialogFragment = ServerErrorDialogFragment()
            serverErrorDialogFragment.arguments = arguments
            return serverErrorDialogFragment
        }
    }
}
